docs/architecture.md

```diff
@@ -112,71 +112,199 @@ The redux store is composed of two `slices` : `documents` and `queries`:
+
+For example:
+```js
+{
+  query1: {
+    definition: {
+      ...
+    },
+    fetchStatus: "loaded",
+    lastFetch: 1716989816939,
+    lastUpdate: 1716989816939,
+    hasMore: false,
+    count: 2,
+    bookmark: "xyz",
+    data: ["docId1", "docId2"] // confusingly named `data`, but only stores ids
+  } 
+}
 ```
 
-After the call to `save` your `TodoListComponent` will be re-rendered 
-with the `New TODO`. 
+#### Fetch status
+
+An important attribute used for the query lifecycle is the `fetchStatus`.
+It can take the following values:
+- `pending`: the query is about to be run for the first time.
+- `loading`: the query is currently running and might return results. From a UX perspective, it is often used by apps to display a spinner.
+- `loaded`: the query had been run and returned results.
+- `failed`: the last query execution failed.
+
+When a query is called for the first time, it is first initialized in the `queries` collection in the store, with the `pending` status.
+
+If the query already exists in the store, its status is checked to ensure that it is not already in a `loading` state.
+
+Then, the query is "loaded", making its status on `loading`.
 
-### How does it works?
+Finally, the query is actually run and the results are retrieved from the database. The status is now `loaded` and some additional information are saved in the query store, such as `lastFetch` and `lastUpdate`.
+If any new data is retrieved by the query, all the documents ids retrieved by the query are then stored in the `data` array. 
 
-When a `query` is resolved, CozyClient dispatches a `receiveQueryResult`
-action for a simple `get` but `receiveMutationResult` when we mutate 
-something. 
+Likewise, the `documents` collection is updated as well with the full documents content.
 
-### Focus on `receiveMutationResult`: 
 
-Our two slices `documents` and `queries` listen actions and do some specific work on
- `isReceivingMutationResult` action.
+#### Auto query update
 
-`documents`: If the `_id` of the mutated document is not present, then we add the document. 
-If the `_id` is already there, then we update the content of the document with the fresh data
-(for now the work is done in extractAndMergeDocument() method). 
+Since there is a link between queries and documents through the ids list in each query object, we need to be able to deal with changes occuring in the documents collection, i.e. when documents are added or deleted.
 
-So if your app is linked to the documents store via getDocumentFromStore() for instance
-your app will have the updated value. 
+For this, cozy-client has a mechanism called "query updater": each time a query is run, all the queries in the store are re-evaluated directly in the store.
+That means that for each query, we take its definition, convert it into a mango predicate, and run it against the store's document collection.
```

Comment:
Maybe we can talk about sift.js here to explain why its important to have compatible query

Reply:
I intentionally avoid mentioning sift. In the previous doc, it was mentioned in a quite technical manner that I found a bit hard to read. Here, I feel like it brings more distraction than value in this particular paragraph, but: you are definitely right that it is worth mentioning sift somewhere. So I'll add a dedicated section.